.github/workflows/acr_deploy_reusable.yml

```diff
@@ -0,0 +1,84 @@
+name: Build and Deploy to ACR Reusable
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+    secrets:
+      ACR_NAME:
+        description: "Name of the Azure container registry to deploy to"
+        required: true
+      ACR_USERNAME:
+        description: "Token name to login to the ACR"
+        required: true
+      ACR_PASSWORD:
+        description: "Token password to login to the ACR"
+        required: true
+      IMAGE_NAME:
+        description: "Name of the docker image to push to the ACR"
+        required: true
+      ARM_CLIENT_ID:
+        required: false
```

Comment:
Perhaps add descriptions so people know why they're optional and when needed (i.e. deploying to staging slots)

Reply:
I thought about this but this shouldn't be something that devs need to understand so didn't bother – just more documentation for me to get wrong. ahaha. we can circle back if needed